### ISSUE TITLE
Check Reviewer 1's comment about solver - make sure addressed

### ISSUE THREAD

**Issue by bpbond** [JGCRI/hector#63]
See @rplzzz 's email too.

**Comment by rplzzz** [comment]
How are we doing on this one?  The equation in question (as best I can render it without math symbols) is:

C(n) = C(n-1) - C(n-1)/tau  + E(n),                                              (1)

This is the finite difference form of the equation 

dC/dt = -C/tau + E.                                                                       (2)

Since we represent E as being piecewise constant (i.e., E = E(n) = constant on the interval t=t(n-1) to t=t(n)), we can solve this equation exactly on each one year solution interval:

C(t) = C0*exp(-t/tau) + E*tau*(1-exp(-t/tau))                                  (3)

If E is given in Gg/yr, then we can just plug in t=1 to get the concentration at the end of the year:

C(year-end) = C0*exp(-1/tau) + E*tau*(1-exp(-1/tau))                 (4)

The Taylor expansion for exp(-1/tau) is 1 - 1/tau + 1/2 1/tau^2 - ..., and we can use that to compare equation (4) to the finite difference version in equation (1):

C(year-end) = C0 - C0/tau + 1/2 C0/tau^2 - ... + E*tau*(1/tau - 1/2 1/tau^2+ ...),   (5)

or:

C(year-end) = C0 - C0/tau + E + 1/tau^2*(C0 - E*tau) + ...                                   (6)

The first three terms are exactly equation (1), and the 1/tau^2 term is the leading-order term in the truncation error (but note that it is actually only O(1/tau) for the emission part).  The referee was objecting to using equation (1) with its associated error when equation (4) is an easy substitute.

**Comment by rplzzz** [comment]
I decided to run a quick comparison between the exact and the finite difference versions of the calculation.  Here are the results from the first year with nonzero concentration for a couple of gasses.  "Concentration" is the current (finite difference) version of the calculation.  "Exact" is what you get using the exact formula, and "Diff" is the fractional difference between the two. 
```
C2F6:
Thu Jan 29 16:11:47 2015:DEBUG:run: date: 1899 concentration: 0.0386091 pptv
Thu Jan 29 16:11:47 2015:DEBUG:run: 	Tau: 10000	Exact Ha: 0.0386071	Diff: -5.00008e-05

HFC23:
Thu Jan 29 16:11:47 2015:DEBUG:run: date: 1930 concentration: 0.0592714 pptv
Thu Jan 29 16:11:47 2015:DEBUG:run: 	Tau: 270	Exact Ha: 0.0591618	Diff: -0.00185299

CFC12:
Thu Jan 29 16:11:47 2015:DEBUG:run: date: 1931 concentration: 0.00445731 pptv
Thu Jan 29 16:11:47 2015:DEBUG:run: 	Tau: 100	Exact Ha: 0.0044351	Diff: -0.00500833 

HFC32:
Thu Jan 29 16:11:47 2015:DEBUG:run: date: 1998 concentration: 0.0093275 pptv
Thu Jan 29 16:11:47 2015:DEBUG:run: 	Tau: 4.9	Exact Ha: 0.00843729	Diff: -0.105509 

CH3Br:
Thu Jan 29 16:11:46 2015:DEBUG:run: date: 1746 concentration: -58.1936 pptv
Thu Jan 29 16:11:46 2015:DEBUG:run: 	Tau: 0.7	Exact Ha: 42.5894	Diff: 2.36639
```
As you would expect, the size of the effect depends on the time scale, tau.  For a relatively short-lived gas like HFC32 the difference is about 10%.  The methyl bromide result is particularly interesting.  Because the decay time scale is less than the one-year time step, C0/tau > C0, and we get a concentration less than zero, which is pretty bad.

Because I didn't track the evolution of the exact solution over multiple integration periods (i.e., each time step I started the exact solution at the last value of the finite-difference solution, not at the last value of the exact solution), it's hard to say from this data what the long-term divergence between the two solutions is.  I'm going to run a calculation using the exact solver and compare the final results from the outputstream.  In any case, the exact solution is in, and I think we should commit it to the master branch.  I'm also going to pull it into the GCAM branch.

**Comment by cahartin** [comment]
Thanks @rplzzz for working through this!  Let me take a closer look and let you know if I have any questions.

**Comment by rplzzz** [comment]
Here's a representative plot using HFC-32 (tau=4.9y):
![image](https://cloud.githubusercontent.com/assets/7209964/5982980/cf320eb6-a896-11e4-8c03-b9df7dae43ef.png)

The difference is subtle, with the exact solution rising a little more slowly on the way up and falling a little more slowly on the way down.  The CH3Br component (not shown) is a little rattier, but even it settles into agreement when concentrations are stable or nearly so.  The big transient at the beginning (which results in negative concentrations in the finite-difference version) is due to the very high starting concentration (H0=157.267 in the included input files).  The equilibrium concentration over most of the 20th century is about 6.4.  Are we sure this starting value is right?

Bottom line is, by the end of the run the difference in total forcing over all components is just 0.001 W/M^2.  So, the difference is small, but the fix was easy.  I recommend we put it in.  I'll allow a little time for further discussion, then commit it unless someone objects.

**Comment by cahartin** [comment]
This is great. I think moving in the direction of an exact solution is best.  

I don't know where those initial halocarbon values came from.  I should look that up or at least supply a reference.

**Comment by cahartin** [comment]
@rplzzz  I say go ahead and commit it.  If you could add some detailed documentation on it in the code that would be great.

**Comment by rplzzz** [comment]
The halocarbon code should  be fixed with the commit referenced above.  I think we should take a look at the initial concentration for CH<sub>3</sub>Br before we close this one out.

**Comment by cahartin** [comment]
Thanks for finding this.  H0 - the pre industrial concentration of CH3Br ( as well as CH3Cl) are emissions and NOT initial concentrations.  I'm trying to track the correct concentrations.

**Comment by rplzzz** [comment]
If those are emissions, then the steady-state concentration should just be E*tau (use equation 4 and take the limit as t->infinity).  That will give you a result in the same units as the emissions, of course, but converting to concentration should be straightforward.

**Comment by cahartin** [comment]
the ini files have been updated with the correct concentration values. 
commit 85eb7a66a0cf04ab308e60529dc857d21b58444f
